src/coreclr/vm/interpexec.cpp

```diff
@@ -934,253 +942,305 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
-            {
-                size_t len = LOCAL_VAR(ip[2], size_t);
-                void* pMemory = NULL;
+                    // clear the valuetype
+                    memset(vtThis, 0, vtSize);
+                    // pass the address of the valuetype
+                    LOCAL_VAR(callArgsOffset, void*) = vtThis;
 
-                if (len > 0)
+                    ip += 5;
+                    goto CALL_INTERP_SLOT;
+                }
+                case INTOP_ZEROBLK_IMM:
+                    memset(LOCAL_VAR(ip[1], void*), 0, ip[2]);
+                    ip += 3;
+                    break;
+                case INTOP_LOCALLOC:
                 {
-                    pMemory = pThreadContext->frameDataAllocator.Alloc(pFrame, len);
-                    if (pMemory == NULL)
+                    size_t len = LOCAL_VAR(ip[2], size_t);
+                    void* pMemory = NULL;
+
+                    if (len > 0)
                     {
-                        // Interpreter-TODO: OutOfMemoryException
-                        assert(0);
+                        pMemory = pThreadContext->frameDataAllocator.Alloc(pFrame, len);
+                        if (pMemory == NULL)
+                        {
+                            // Interpreter-TODO: OutOfMemoryException
+                            assert(0);
+                        }
+                        if (pMethod->initLocals)
+                        {
+                            memset(pMemory, 0, len);
+                        }
                     }
-                    if (pMethod->initLocals)
+
+                    LOCAL_VAR(ip[1], void*) = pMemory;
+                    ip += 3;
+                    break;
+                }
+                case INTOP_GC_COLLECT:
+                {
+                    // HACK: blocking gc of all generations to enable early stackwalk testing
+                    // Interpreter-TODO: Remove this
                     {
-                        memset(pMemory, 0, len);
+                        pInterpreterFrame->SetTopInterpMethodContextFrame(pFrame);
+                        GCX_COOP();
+                        GCHeapUtilities::GetGCHeap()->GarbageCollect(-1, false, 0x00000002);
                     }
+                    ip++;
+                    break;
                 }
-
-                LOCAL_VAR(ip[1], void*) = pMemory;
-                ip += 3;
-                break;
-            }
-            case INTOP_GC_COLLECT: {
-                // HACK: blocking gc of all generations to enable early stackwalk testing
-                // Interpreter-TODO: Remove this
+                case INTOP_THROW:
                 {
-                    pInterpreterFrame->SetTopInterpMethodContextFrame(pFrame);
-                    GCX_COOP();
-                    GCHeapUtilities::GetGCHeap()->GarbageCollect(-1, false, 0x00000002);
+                    CONTEXT exceptionContext;
+                    ClrCaptureContext(&exceptionContext);
```

Comment:
How much do we actually use in this context? I assume that it is just a tiny part. Would it make sense to capture just what's actually used?

Reply:
Ah ok, I keep forgetting about the floating-point registers special case.

Reply:
> I guess we could change it to capture just the non-volatile floating point registers and PC/FP/SP (for all the EH even without interpreter).

We need to deal with the non-volatile floating-point registers on Windows x64 only. Is that right? It may make sense to save everything on Windows x64 only.

In any case, this is not something to address as part of this change.

Reply:
It is also needed for ARM64 on all platforms, the D8..D15 are non-volatile there. Similar for RISCV64 and LOONGARCH64 that also have non-volatile floating point registers. And ARM32 too.
Linux x64 is the only one that doesn't have any non-volatile floating point registers.